# Make Prediction Request

In [1]:
import json
import requests
import tensorflow as tf

def create_example(features_dict):
    # Fitur dikodekan sebagai string -> TF Example
    feature = {
        key: tf.train.Feature(bytes_list=tf.train.BytesList(value=[value.encode()]))
        for key, value in features_dict.items()
    }
    example_proto = tf.train.Example(features=tf.train.Features(feature=feature))
    return example_proto.SerializeToString()

# Contoh input (harus valid dan sama formatnya dengan data training)
example_dict = {
    "cap-shape": "x",
    "cap-surface": "s",
    "cap-color": "n",
    "bruises": "t",
    "odor": "p",
    "gill-attachment": "f",
    "gill-spacing": "c",
    "gill-size": "n",
    "gill-color": "k",
    "stalk-shape": "e",
    "stalk-root": "e",
    "stalk-surface-above-ring": "s",
    "stalk-surface-below-ring": "s",
    "stalk-color-above-ring": "w",
    "stalk-color-below-ring": "w",
    "veil-type": "p",
    "veil-color": "w",
    "ring-number": "o",
    "ring-type": "p",
    "spore-print-color": "k",
    "population": "s",
    "habitat": "u"
}

# Serialisasi ke TF Example
serialized_example = create_example(example_dict)

# Kirim ke TensorFlow Serving
endpoint = "http://localhost:8080/v1/models/mushroom-model:predict"
headers = {"content-type": "application/json"}
data = json.dumps({"instances": [serialized_example.decode("ISO-8859-1")]})  # decode to string

response = requests.post(endpoint, data=data, headers=headers)

# Hasil prediksi
print("Raw response:", response.json())
predicted_prob = response.json()["predictions"][0][0]
predicted_label = 1 if predicted_prob >= 0.5 else 0

print(f"Predicted Label: {predicted_label} | Probability: {predicted_prob:.4f}")


Raw response: {'predictions': [[0.998838544]]}
Predicted Label: 1 | Probability: 0.9988


Jamur tersebut diprediksi sebagai Poisonous (beracun) dengan tingkat keyakinan 99.88%.